### PR TITLE
docs: adds collapsible elements to the readme to minify their length

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

- minify readme footprint
- fix(docs): fix various types
- feat(docs): add TOC

### Close Issue(s)

-

### CODE REVIEW

**Comment by provinzkraut** [comment]
We’ll need to further customise the PyPI readme then since afaik it won’t display these properly. We should be able to just extract the content from the sections though I think.

**Comment by JacobCoffee** [comment]
@provinzkraut I believe it works since its standard HTML

https://pypi.org/project/coffeetests/0.1.0/

**Comment by provinzkraut** [comment]
> @provinzkraut I believe it works since its standard HTML
> 
> https://pypi.org/project/coffeetests/0.1.0/

Cool!

File: README.md [hunk 1]
```diff
@@ -35,6 +35,27 @@ its features!
 Additionally, the [Litestar fullstack repository](https://github.com/litestar-org/litestar-fullstack)
 can give you a good impression how a fully fledged Litestar application may look.
 
+<details>
```
Comment (provinzkraut) [inline README.md:38]: Do we need that? GH already includes a toc. Maybe just a matter of personal taste, but if we do have a toc, I think it should be visible directly

Comment (FHU-yezi) [reply README.md:38]: Maybe for mobile guys? GitHub App does not have TOC.

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**